src/main.cpp

```diff
@@ -115,748 +160,765 @@ int main ( int argc, char** argv )
+            // Show analyzer console -----------------------------------------------
+            // Undocumented debugging command line argument: Show the analyzer
+            // console to debug network buffer properties.
+            if ( GetFlagArgument ( argv,
+                                   i,
+                                   "--showanalyzerconsole", // no short form
+                                   "--showanalyzerconsole" ) )
+            {
+                bShowAnalyzerConsole = true;
+                qInfo() << "- show analyzer console";
+                CommandLineOptions << "--showanalyzerconsole";
+                ClientOnlyOptions << "--showanalyzerconsole";
+                continue;
+            }
 
-    if ( bIsClient )
-    {
-        if ( ServerOnlyOptions.size() != 0 )
-        {
-            qCritical() << qUtf8Printable ( QString ( "%1: Server only option(s) '%2' used.  Did you omit '--server'?" )
-                                                .arg ( argv[0] )
-                                                .arg ( ServerOnlyOptions.join ( ", " ) ) );
-            exit ( 1 );
+            // Unknown option ------------------------------------------------------
+            throw CErrorExit ( qUtf8Printable ( QString ( "%1: Unknown option '%2' -- use '--help' for help" ).arg ( argv[0] ).arg ( argv[i] ) ) );
         }
 
-        // mute my own signal in personal mix is only supported for headless mode
-        if ( bUseGUI && bMuteMeInPersonalMix )
-        {
-            bMuteMeInPersonalMix = false;
-            qWarning() << "Mute my own signal in my personal mix is only supported in headless mode.";
-        }
+        // Dependencies ------------------------------------------------------------
 
-        // adjust default port number for client: use different default port than the server since
-        // if the client is started before the server, the server would get a socket bind error
-        if ( !bCustomPortNumberGiven )
+#if ( defined( SERVER_BUNDLE ) && defined( Q_OS_MACX ) ) || defined( SERVER_ONLY )
+        // if we are on MacOS and we are building a server bundle or requested build with serveronly, start Jamulus in server mode
+        if ( bIsClient )
         {
-            iPortNumber += 10; // increment by 10
-            qInfo() << qUtf8Printable ( QString ( "- allocated port number: %1" ).arg ( iPortNumber ) );
+            bIsClient = false;
+            qInfo() << "- Starting in server mode by default (due to compile time option)";
         }
-    }
-    else
-    {
-        if ( ClientOnlyOptions.size() != 0 )
+#endif
+
+#ifdef HEADLESS
+        Q_UNUSED ( bStartMinimized )       // avoid compiler warnings
+        Q_UNUSED ( bShowComplRegConnList ) // avoid compiler warnings
+        Q_UNUSED ( bShowAnalyzerConsole )  // avoid compiler warnings
+        Q_UNUSED ( bMuteStream )           // avoid compiler warnings
+
+        if ( bUseGUI )
```

Comment:
Don't change this. In particular it's not a good idea to change error messages unless it's absolutely necessary - they show up in documentation.